### ISSUE TITLE
messages not in order

### ISSUE THREAD

**Issue by ghost** [jlippold/WifiSMS#5]
me again :)
when having a "fast" conversation, as in sms coming in and going out every 2 secs or so,
the messages' order is incorrect.
i _think_ this is because of the "check interval", but the bug is actually: if it fetches messages every N secs, it first fetches incomings and then outgoings, and dont sort them by time, so if you have a very long conv. within the timeframe of the check-interval, all of the incomings will be listed and then the outgoings...

**Comment by treason** [comment]
the way it works is that outgoing messages are queued by the browser, so if you send 10 SMS messages in succession, it sends the first, wait for the sent successful, then does the next. So the iPhone doesnt know the next message is even in queue until it gets there. I used to set the sent time from the browser and not the iphone, but if your computer clock is off it would look screwy. 

The proper solution is to send immediately to the iphone and not queue on the browser. The problem here is knowing what was sent when the "Sent Successful" callback is reached. 

So, long story short, it has nothing to do with the refresh interval or anything. I just need to find a better way to handle it.

**Comment by ecnerd** [comment]
i'm having the same problem, but i noticed something else.

The message order on the native message app is CORRECT. So is it possible to just query the sms.db every once in a while to make any corrections?

**Comment by ghost** [comment]
same, it shows up fine in Messages/biteSMS on the iPhone, but not on WifiSMS.

**Comment by Spidored** [comment]
I got the same, so like precedently said, the order inside SMS app are correct, but in the browser it really fail.

When i send a message, eg now it's 02:13, the sent time in the app will be 02:13, then when the person will reply, it will show that he replied before i sent eg : 02:10. I don't know where's the problem exactly but it's some kind of mess.

PS : Inside my SMS app, the time is wrong too but the order are correct. maybe sorting the SMS by ID in the sqlite instead of sorting by date.

**Comment by treason** [comment]
that makes sense Spidored, I could just sort by row ID... would that solve the problem for everyone?

**Comment by Spidored** [comment]
i think it could be great like this, the rowid will always be the good order instead of time.

**Comment by treason** [comment]
will be on 1.0.5

**Comment by treason** [comment]
Would you try 1.0.5 and confirm fixed? https://github.com/treason/WifiSMS/blob/master/Build/deb/WifiSMS-NOSBS-1.0.5.deb?raw=true
